File: android/feature/aiconnect/src/main/java/com/enterprise/feature/aiconnect/domain/entities/ChatState.kt
package com.enterprise.feature.aiconnect.domain.entities

import com.enterprise.core.domain.entities.DomainError
import com.enterprise.core.domain.entities.UCPMessage

/** Chat state for the AI Connect MVI pattern. */
data class ChatState(
    val messages: List<UCPMessage> = emptyList(),
    val isConnected: Boolean = false,
    val isStreaming: Boolean = false,
    val currentStreamText: String = "",
    val error: DomainError? = null
)

File: ios/EnterpriseMobileApp/Sources/Feature/AIConnect/Domain/Entities/ChatState.swift
import Foundation
import CoreDomain

/// Chat feature state for the AI Connect tab.
public struct ChatState: Equatable, Sendable {
    public var messages: [UCPMessage]
    public var isConnected: Bool
    public var isStreaming: Bool
    public var currentStreamText: String
    public var error: DomainError?

    public init(
        messages: [UCPMessage] = [],
        isConnected: Bool = false,
        isStreaming: Bool = false,
        currentStreamText: String = "",
        error: DomainError? = nil
    ) {
        self.messages = messages
        self.isConnected = isConnected
        self.isStreaming = isStreaming
        self.currentStreamText = currentStreamText
        self.error = error
    }
}
